File: examples/proxy/python/http_server.py
from flask import Flask, request
from werkzeug.serving import run_simple

import uuid
import proxy

service = {
    "name": "go.micro.srv.greeter",
    "nodes": [{
        "id": "go.micro.srv.greeter-" + str(uuid.uuid4()),
        "address": "127.0.0.1",
        "port": 4000,
    }],
}

app = Flask(__name__)

@app.route('/greeter', methods=['POST'])
def hello_world():
    name = request.values['name']
    if len(name) == 0:
      name = 'World'
    return 'Hello ' + name + '!'

if __name__ == '__main__':
    proxy.register(service)
    run_simple('localhost', 4000, app)
    proxy.deregister(service)

File: examples/proxy/ruby/http_server.rb
require './proxy'
require 'securerandom'
require 'webrick'

$service = {
  "name" => "go.micro.srv.greeter",
  "nodes" => [{
    "id" => "go.micro.srv.greeter-" + SecureRandom.uuid,
    "address" => "localhost",
    "port" => 4000
  }]
}

trap 'INT' do
  deregister($service)
  exit
end

# create server
server = WEBrick::HTTPServer.new :Port => 4000

# serve method Say.Hello
server.mount_proc '/greeter' do |req, res|
  res.body = "Hello #{req.query['name']}!"
end

# register service
register($service)

# start the server and block
server.start
